packages/jaeger-ui/src/model/ddg/transformTracesToPaths.tsx

```diff
@@ -35,22 +38,33 @@ function transformTracesToPaths(
           spanMap.set(span.spanID, span);
           return !span.hasChildren;
         })
+
         .forEach(leaf => {
-          const spans = spanAncestorIds(leaf).map(id => {
+          const ancestors = spanAncestorIds(leaf).reverse();
+          ancestors.push(leaf.spanID);
+          const spans = ancestors.reduce((reducedSpans: Span[], id: string): Span[] => {
             const span = spanMap.get(id);
             if (!span) throw new Error(`Ancestor spanID ${id} not found in trace ${traceID}`);
-            return span;
-          });
-          spans.reverse();
-          spans.push(leaf);
-
-          const path: TDdgPayloadEntry[] = spans
-            .filter(span => span.tags.find(({ key, value }) => key === 'span.kind' && value === 'server'))
-            .map(({ processID, operationName: operation }) => ({
-              service: data.processes[processID].serviceName,
-              operation,
-            }));
-
+            if (reducedSpans.length > 0) {
+              const headSpan = reducedSpans[reducedSpans.length - 1];
+              // Transition inside the same service ServiceA -> ServiceA
+              if (headSpan.processID === span.processID) {
```

Comment:
@rubenvp8510 sorry I couldn't get to this sooner

If I'm reading this correctly, it captures every change in `processID`. Then, if there isn't a change in `processID` and the current span is of kind `server`:
1) a parent `client` span that was a different process than _its_ parent span gets replaced with the `server` span
2) a parent `server` span is kept and the current span is added.

I'm wondering if we need to do the replacement in case 1, or if it would be better to simply add any span of kind `server` or any span with a different `processID` than the previous reduced span.
i.e.:
```
if (headSpan.processID !== span.processID || isKindServer(span) {
  reducedSpans.push(span);
}
```
The only case this is different is if `process a` calls `process b` which emits a `client` span as its first span, which is the parent of a `server` span in the same process. IMO this case is strange, possibly caused by bad instrumentation, and I'd prefer to surface the scenario to potentially fix it rather than bury it.

@yurishkuro @vprithvi I'd like to hear your opinions on the two approaches to handling this case.

Reply:
@rubenvp8510 Update: I have discussed the two approaches with Yuri and Prithvi and we agree to not replace the ingress client spans.

Here's an example, which, for brevity, I will represent a single span as `processID,operationName,[s]erver/[c]lient`
e.g.:
```
{
  processID: 0,
  operationName: 'a',
  span.kind: 'client',
}
```
would be `0,a,c`

original:
`0,a,c 1,a,s 1,b,c 1,c,s, 2,a,c, 2,b,s 2,c,c`
would be reduced to
`0,a,c 1,a,s 1,c,s 2,a,c 2,b,s` by my proposed method
versus
`0,a,c 1,a,s 1,c,s 2,b,s` in the PR

The differences is that the client ingress span (`2,a,c`) is not dropped.

Reply:
Hi @everett980 I think it is OK not loses the ingress client span. it just wasn't clear to me what would be the correct way of derivative the paths on those cases, but now it makes more sense. I'll do the pertinent changes.

Reply:
I did the changes, this is ready for another review.

Thank you!